### PR TITLE
This PR adds PhysX examples that are equivalent to all existing examples in aframe-physics system

### PR DESCRIPTION
**Pull request by diarmidmackenzie**
Most of these examples could be implemented without core code changes.

There are just a couple of bug fixes
- rework of perf optimization, which turned out to have broken constraints
- fix a typo siffness-> stiffness, though even with this fix, springs still don't work :-(

Most examples work fine.  The only exceptions are:
- we have issues with scale.   Exhibited both in the sandbox example, and in the fact that I had to remove all scaling from the constraints example.
- springs example doesn't work.  Soft linear constraints just don't seem to work correctly, and I don'y yet know why.

Those two problems might not be fast to solve, so I propose merging this PR as is, and raising separate issues to track those problems.

### CODE REVIEW

**Comment by diarmidmackenzie** [comment]
While this PR is under review, I've publsihed the source branch at GitHub Pages here:
https://diarmidmackenzie.github.io/physx/examples/

But note that links don't work - you'll need these instead:

https://diarmidmackenzie.github.io/physx/examples/basic/index.html
https://diarmidmackenzie.github.io/physx/examples/sandbox/index.html
https://diarmidmackenzie.github.io/physx/examples/compound/index.html
https://diarmidmackenzie.github.io/physx/examples/constraints/index.html
https://diarmidmackenzie.github.io/physx/examples/materials/index.html
https://diarmidmackenzie.github.io/physx/examples/spring/index.html
https://diarmidmackenzie.github.io/physx/examples/stress/index.html
https://diarmidmackenzie.github.io/physx/examples/ttl/index.html
https://diarmidmackenzie.github.io/physx/examples/pinboard/physx.html
https://diarmidmackenzie.github.io/physx/examples/pinboard/ammo-vs-physx.html

**Comment by kylebakerio** [comment]
exciting! fantastic work, will try to look into this soon!

**Review by vincentfretin — COMMENTED** [review]
This is great!
I added some comments in the code. I also looked at each example.

On https://diarmidmackenzie.github.io/physx/examples/spring/index.html
The "view code" text is on the left and not in the white circle on the right.

https://diarmidmackenzie.github.io/physx/examples/pinboard/ammo-vs-physx.html
I wasn't sure which one was PhysX. The html file name is ammo-vs-physx in this order and in the description it says physx and ammo. I had to look at the html source to confirm PhysX is on the left. You should specify in the description that PhysX is on the left, and ammo on the right.

File: examples/materials/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+             environment
+             physx="autoLoad: true; delay: 1000; useDefaultScene: false">
+      <a-entity camera look-controls wasd-controls position="0 1.6 0">
+        <a-entity cursor
+                  raycaster="objects:[physx-force-pushable]"
+                  position="0 0 -0.5"
+                  geometry="primitive: circle; radius: 0.01; segments: 4;"
+                  material="color: #FF4444; shader: flat"></a-entity>
+      </a-entity>
+      <a-entity id="left-hand" ammo-body="type: kinematic; emitCollisionEvents: true" ammo-shape="type: sphere; fit: manual; sphereRadius: 0.02;"
```
[the remark this answers is not in the document]
Comment (diarmidmackenzie) [reply examples/materials/index.html]: Thanks - replaced with physx

File: examples/components/grab.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    // the entity is automatically transformed into a kinematic entity.
+    // To avoid triggering this (we want to grab using constraints, and leave the
+    // body as dynamic), we use a non-clashing name for the state we set on the entity when
+    // grabbing it.
+    this.GRABBED_STATE = 'grabbed-dynamic';
+
+    this.grabbing = false;
+    this.hitEl =      /** @type {AFRAME.Element}    */ null;
+    this.physics =    /** @type {AFRAME.System}     */ this.el.sceneEl.systems.physics;
+    this.constraint = /** @type {CANNON.Constraint} */ null;
```
[the remark this answers is not in the document]
Comment (diarmidmackenzie) [reply examples/components/grab.js]: Deleted line (not used any more)

File: examples/components/force-pushable.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ */
+AFRAME.registerComponent('physx-force-pushable', {
+  schema: {
+    force: { default: 10 }
+  },
+  init: function () {
+
+    this.pStart = new THREE.Vector3();
+    this.sourceEl = this.el.sceneEl.querySelector('[camera]');
+    this.el.addEventListener('click', this.forcePushPhysX.bind(this));
```
[the remark this answers is not in the document]
Comment (diarmidmackenzie) [reply examples/components/force-pushable.js]: Fixed.

**Pull request merged**